dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -807,6 +809,59 @@ public void createDirectory(String path, CreateDirectoryPOptions options)
     }
   }
 
+  @Override
+  public boolean exists(String path, ExistsPOptions options) {
+    long syncIntervalMs = options.hasCommonOptions()
+        ? (options.getCommonOptions().hasSyncIntervalMs()
+        ? options.getCommonOptions().getSyncIntervalMs() : -1) :
+        -1;
+
+    Optional<DoraMeta.FileStatus> status = mMetaManager.getFromMetaStore(path);
+    if (status.isPresent()) { // metadata exists.
+      if (syncIntervalMs >= 0) {
+        if (System.nanoTime() - status.get().getTs() < syncIntervalMs * Constants.MS_NANO) {
+          // metadata exists and is not out-of-date.
+          return true;
+        }
+      } else {
+        // metadata exists and client does not require to check cache validity.
+        return true;
+      }
+    }
+
+    // TODO(Yimin) Load/reload metadata from UFS
+    try {
+      return mUfs.exists(path);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void setAttribute(String path, SetAttributePOptions options) {
```

Comment:
there's a `recursive` option for `setAttribute` to set the same set of attributes to all files under a directory. I don't think we support doing that under Dora, so either we remove that option, or leave a todo to remove that option.

Reply:
We don't support recursive at this moment.
I just removed the checking of this option here.
Maybe it's better to add comments about this.

Reply:
Added comments.